wakuv2/waku.go

```diff
@@ -1318,7 +1166,28 @@ func (w *Waku) Start() error {
 
 	go w.sendQueue.Start(w.ctx)
 
-	go w.checkIfMessagesStored()
+	if w.cfg.EnableStoreConfirmationForMessagesSent {
+		w.messageSentCheck = publish.NewMessageSentCheck(w.ctx, w.node.Store(), w.node.Timesource(), w.logger)
+		go w.messageSentCheck.CheckIfMessagesStored()
+		go func() {
+			for {
+				select {
+				case <-w.ctx.Done():
+					return
+				case hash := <-w.messageSentCheck.MessageStoredChan:
```

Comment:
Why not return hash and state as a combo rather than having 2 channels?
Any specific reason

Reply:
Need to create a new type to differentiate the EventEnvelopeSent/MessageStored and EventEnvelopeExpired/MessageExpired, and convert between types in status-go and go-waku, which seems not better than having two channels.